src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -962,38 +1232,99 @@ function AnalyticsService:GetRemoteConfig(remote: RemoteConfig): { [any]: any }
 		elseif not remote.defaultValue then
 			return reject("AnalyticsService.GetRemoteConfig - remote.defaultValue is required")
 		end
-		
+
 		if not self._enabled then
 			return resolve(remote.defaultValue)
 		end
-		
+
 		local player: Player? = remote.player
-		local context: any = self:_getServerPsuedoPlayer()
-		local server: any = not player and HttpApi:initRequest(
-			self._options.gameKey,
-			self._options.secretKey,
-			self._options.build,
-			context.PlayerData,
-			""
-		)
-		
+		local context: ServerPsuedoPlayer = self:_getServerPsuedoPlayer()
+		local server = if player == nil
+			then -- Using Luau conditional expression so type can be inferred https://luau-lang.org/syntax#if-then-else-expressions
+				GameAnalytics.HttpApi:initRequest(
+					self._options.gameKey,
+					self._options.secretKey,
+					self._options.build,
+					context.PlayerData,
+					""
+				)
+			else nil
+
 		if server and server.statusCode >= 9 then
 			for _, config in (server.body.configs or {}) do
 				if config.key == remote.name then
 					return resolve(config.value)
 				end
 			end
 		end
-		
+
 		if player and not GameAnalytics:isRemoteConfigsReady(player.UserId) then
 			return resolve(remote.defaultValue)
 		end
-		
-		return resolve(player and GameAnalytics:getRemoteConfigsValueAsString(player.UserId, {
+
+		resolve(player and GameAnalytics:getRemoteConfigsValueAsString(player.UserId, {
```

Comment:
```suggestion
		return resolve(player and GameAnalytics:getRemoteConfigsValueAsString(player.UserId, {
```
The linter/type checker complains if there's a return type defined for a function, but the promise is resolved/rejected without returning the function (For some reason)? It's an invalid error since we know we're returning a promise, but the code checker flags it for some reason.